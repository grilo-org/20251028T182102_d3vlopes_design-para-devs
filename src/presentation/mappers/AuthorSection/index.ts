import { AuthorSectionAPI } from '@/api/models/Sections';
import { AuthorSectionModel } from '@/domain/models/Sections';

export class AuthorSectionMapper {
	static toDomain(api: AuthorSectionAPI): AuthorSectionModel {
		return {
			id: api.sectionID,
			description: api.description,
			name: api.name,
			expertise: api.expertise,
			avatar: {
				src: api.photo.url,
				alt: String(api.photo.imageDescription),
			},
			socials: api.socials.map((social) => {
				return {
					id: social.id,
					name: social.name,
					href: social.url,
					src: social.icon.url,
					alt: String(social.icon.imageDescription),
				};
			}),
		};
	}
}
