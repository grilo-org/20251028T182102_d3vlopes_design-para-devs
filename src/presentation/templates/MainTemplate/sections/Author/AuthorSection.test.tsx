import { render, screen } from '@testing-library/react';

import { authorSectionMock as mock } from './AuthorSection.mock';

import { AuthorSection } from '.';

jest.mock('@/presentation/components/molecules/Avatar', () => ({
	Avatar: () => <div data-testid="mock-avatar" />,
}));

jest.mock('@/presentation/components/molecules/SocialIcons', () => ({
	SocialIcons: () => <div data-testid="mock-social-icons" />,
}));

const { getByRole, getByTestId, getByText } = screen;

describe('<AuthorSection />', () => {
	beforeEach(() => render(<AuthorSection {...mock} />));

	it('should contain correct section id', () => {
		const { container } = render(<AuthorSection {...mock} />);

		const section = container.getElementsByTagName('section')[0];

		expect(section).toHaveAttribute('id', mock.id);
	});

	it('should render Avatar', () => {
		const avatar = getByTestId('mock-avatar');

		expect(avatar).toBeInTheDocument();
	});

	it('should render name', () => {
		const authorName = getByRole('heading', { name: mock.name });

		expect(authorName).toBeInTheDocument();
	});

	it('should render expertise', () => {
		const authorExpertise = getByRole('heading', {
			name: mock.expertise,
		});

		expect(authorExpertise).toBeInTheDocument();
	});

	it('should render SocialIcons', () => {
		const socialIcons = getByTestId('mock-social-icons');

		expect(socialIcons).toBeInTheDocument();
	});

	it('should render description', () => {
		const description = getByText(mock.description);

		expect(description).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		const { container } = render(<AuthorSection {...mock} />);

		expect(container).toMatchSnapshot();
	});
});
