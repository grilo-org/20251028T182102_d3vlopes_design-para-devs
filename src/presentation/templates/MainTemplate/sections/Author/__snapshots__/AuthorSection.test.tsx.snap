// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthorSection /> should match snapshot 1`] = `
<div>
  <section
    class="flex flex-col justify-center items-center py-14 max-w-[78.4287rem] lg:flex-row lg:justify-start lg:m-auto lg:px-[5rem] lg:gap-[2.9375rem] lg:pt-16 lg:pb-[5.625rem] xl:px-0"
    id="author"
  >
    <div
      data-testid="mock-avatar"
    />
    <div
      class="flex flex-col items-center lg:items-start"
    >
      <div
        class="flex flex-col mt-4 mb-6 lg:mt-0 lg:mb-8"
      >
        <h2
          class="font-heading font-semibold text-8xl text-complementary-white text-center lg:text-left"
        >
          Leandro Lopes
        </h2>
        <h3
          class="font-body text-base text-neutral-100 lg:text-4xl"
        >
          Desenvolvedor Full Stack | UX/UI Designer
        </h3>
      </div>
      <div
        data-testid="mock-social-icons"
      />
      <p
        class="font-body text-base mt-4 text-center text-neutral-50 px-7 md:max-w-[35rem] lg:mt-8 lg:text-left lg:px-0 xl:max-w-[51.78rem] xl:text-xxl xl:leading-[170%]"
      >
        Olá, me chamo Leandro Lopes, sou desenvolvedor Full Stack e UX/UI Designer com mais de 4 anos de experiência e fundador do Union Developers, uma empresa sem fins lucrativos com o objetivo de ajudar desenvolvedores júnior. Sou apaixonado por código, criação e compartilhar conhecimento.
      </p>
    </div>
  </section>
</div>
`;
